assets/src/edit-story/components/panels/size.js

```diff
@@ -18,101 +18,218 @@
+        const hasHeightButNotWidth = newHeight === '' || newWidth === '';
+        const hasWidthButNotHeight = newHeight === '' && newWidth === '';
+
+        if (lockRatio && hasHeightButNotWidth && !hasWidthButNotHeight) {
+          const ratio = oldWidth / oldHeight;
+          if (newWidth === '') {
+            update.width = dataPixels(newHeight * ratio);
+          } else if (newHeight === '') {
+            update.height = dataPixels(newWidth / ratio);
+          }
         }
+        return update;
+      });
+      if (evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
       }
-      return { width: newWidth, height: newHeight };
-    });
-    evt.preventDefault();
-  };
+    },
+    [lockRatio, onSetProperties, state]
+  );
   return (
     <SimplePanel
       name="size"
-      title={__('Size', 'web-stories')}
-      onSubmit={handleSubmit}
+      title={__('Size & position', 'web-stories')}
+      onSubmit={updateProperties}
     >
-      <InputGroup
-        label={__('Width', 'web-stories')}
-        value={state.width}
-        isMultiple={width === ''}
-        onChange={(value) => {
-          const ratio = width / height;
-          const newWidth =
-            isNaN(value) || value === '' ? '' : parseFloat(value);
-          setState({
-            ...state,
-            width: newWidth,
-            height:
-              height !== '' && typeof newWidth === 'number' && lockRatio
-                ? dataPixels(newWidth / ratio)
-                : height,
-          });
-        }}
-        postfix={_x('px', 'pixels, the measurement of size', 'web-stories')}
-        disabled={isFill}
-      />
-      <InputGroup
-        label={__('Height', 'web-stories')}
-        value={state.height}
-        isMultiple={height === ''}
-        onChange={(value) => {
-          const ratio = width / height;
-          const newHeight =
-            isNaN(value) || value === '' ? '' : parseFloat(value);
-          setState({
-            ...state,
-            height: newHeight,
-            width:
-              width !== '' && typeof newHeight === 'number' && lockRatio
-                ? dataPixels(newHeight * ratio)
-                : width,
-          });
-        }}
-        postfix={_x('px', 'pixels, the measurement of size', 'web-stories')}
-        disabled={isFill}
-      />
-      <InputGroup
-        type="checkbox"
-        label={__('Keep ratio', 'web-stories')}
-        value={lockRatio}
-        isMultiple={false}
-        onChange={(value) => {
-          setLockRatio(value);
-        }}
-        disabled={isFill}
-      />
+      {/** Position */}
+      <Row expand>
+        <BoxedNumeric
+          prefix={_x('X', 'The X axis', 'web-stories')}
+          value={state.x}
+          isMultiple={x === ''}
+          onChange={(value) =>
+            setState({
+              ...state,
+              x: isNaN(value) || value === '' ? '' : parseFloat(value),
+            })
+          }
```

Comment:
This handler is used many times - maybe create a generic handler above as:

```js
const handleNumberChange = useCallback((property) => (value) => setState((state) => ({
   ...state,
   [property]: isNaN(value) || value === '' ? '' : parseFloat(value),
}))
```

And then where used simply do:

```js
onChange={handleNumberChange('x')}
```

Reply:
Done.